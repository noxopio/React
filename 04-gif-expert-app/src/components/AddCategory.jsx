import { useState } from "react"

// eslint-disable-next-line react/prop-types
export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newValue=(inputValue.trim().toUpperCase());
        if(newValue.length <=1) return;
        // setCategories( categories=>[ inputValue,...categories])
        onNewCategory(newValue)
        setInputValue('')
    }
    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

export default AddCategory
