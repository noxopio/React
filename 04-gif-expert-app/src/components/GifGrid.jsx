

// eslint-disable-next-line react/prop-types
const GifGrid = ({ category }) => {
  return (<><h3>{category}</h3><p>hell is empty</p></>
  )
}

export default GifGrid