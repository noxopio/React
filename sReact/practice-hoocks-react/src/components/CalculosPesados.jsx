/* eslint-disable react-hooks/rules-of-hooks */
import { useMemo, useState } from "react";
export const CalculosPesados = () => {
    const [list, setList] = useState([2, 3, 4, 5, 6, 8, 9, 10, 11])
    const [show, setShow] = useState(true)

    const getCalculo =(list) => useMemo( () => {
        console.log('calculando');
        return list.reduce((acc, el) => acc * el)
    },[list]);
    

    const addNum=()=>{

        setList([...list,list[list.length-1]+1])
    }

    return (
        <>
            <h1>Calculo</h1>
            <p>{getCalculo(list)}</p>


            <button className="btn btn-primary" onClick={()=>setShow (!show)}>{show ? 'SHOW' : 'HIDE'}</button>
        <button className="btn btn-primary" onClick={()=>addNum()}  > Agregar</button>
        </>
    )
}
