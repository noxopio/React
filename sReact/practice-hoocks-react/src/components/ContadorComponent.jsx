import { useCounter } from "../hooks/useCounter"

export const ContadorComponent = () => {
const {contador, incrementar, decrementar, resetear} =useCounter(0)

    useCounter(0)
    return (
        <>
            <h1>Contador:{contador}</h1>
            <button className="btn btn-primary" onClick={() => incrementar(1)}>+1</button>
            <button className="btn btn-danger" onClick={() => resetear()}> resetear </button>
            <button className="btn btn-primary" onClick={() => decrementar(1,false)}> -1</button>
        </>
    )
}
