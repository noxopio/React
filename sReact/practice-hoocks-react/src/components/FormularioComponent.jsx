
import { useEffect, useRef } from "react"
import { useForm } from "../hooks/useForm"
export const FormularioComponent = () => {

    const focusRef = useRef();
    const initalForm = {
        userName: '',
        email: '',
        password: ''
    }
    const { userName, email, password, onInputChange } = useForm(initalForm)
    // const {userName,email,password}=formState
    // const [formstate, setFormstate] = useState({
    //     userName: '',
    //     email: '',
    //     password: ''
    // })
    // const { userName, email, password } = formstate
    // const oninputChange = ({ target }) => {
    //     const { name, value } = target
    //     setFormstate({ ...formState, [name]: value })
    // }
    const onSubmit = (event) => {
        event.preventDefault()
        console.log(userName, email, password);
    }
    useEffect(() => {
        focusRef.current.focus()
    }, [])

    return (
        <form onSubmit={onSubmit}>
            <div className="mb-3">
                <label
                    htmlFor="userName"
                    className="form-label"
                >
                    Username
                </label>
                <input
                    type="text"
                    className="form-control"
                    name="userName"
                    placeholder="User name"
                    value={userName}
                    onChange={onInputChange}
                />
            </div>
            <div className="mb-3">
                <label
                  
                    htmlFor="email"
                    className="form-label">
                    Email address
                </label>
                <input
                    type="email"
                    className="form-control"
                    name="email"
                    aria-describedby="emailHelp"
                    placeholder="Email"
                    value={email}
                    onChange={onInputChange}
                    ref={focusRef}

                />
                <div name="emailHelp"
                    className="form-text">Well never share your email with anyone else.</div>
            </div>
            <div className="mb-3">
                <label
                    htmlFor="password"
                    className="form-label">Password</label>
                <input
                    type="password"
                    className="form-control"
                    name="password"
                    placeholder="Password"
                    value={password}
                    onChange={onInputChange}
                />
            </div>
            {/* <div className="mb-3 form-check">
          <input type="checkbox" className="form-check-input" name="exampleCheck1"/>
          <label className="form-check-label" htmlFor="exampleCheck1">Check me out</label>
        </div> */}
            <button type="submit" className="btn btn-primary" >Submit</button>
        </form>
    )
}


