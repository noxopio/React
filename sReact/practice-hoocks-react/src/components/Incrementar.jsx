import React from 'react'

import PropTypes from 'prop-types';

export const Incrementar = React.memo(({ incrementar }) => {
    console.log('me renderize');
    
    return (
        <button onClick={() => incrementar(10)}>+1</button>
    );
});

// Incrementar.displayName = 'Incrementar';

// Incrementar.propTypes = {
//     incrementar: PropTypes.func.isRequired,
// };
