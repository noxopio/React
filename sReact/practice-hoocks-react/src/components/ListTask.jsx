import { useForm } from "../hooks/useForm"

const initilState = [{
    id: new Date().getTime(),
    tarea: 'Explicar React',
    finalizada: false
}]

const nuevatarea = {
    id: 2,
    tarea: 'Explicar React',
    finalizada: false
}

// const addtarea = {
//     type: '[TAREAS]agregar tarea',
//     payload: nuevatarea
// }
const deltarea = {
    type: '[TAREAS]eliminar tarea',
    payload: nuevatarea
}

const tareasReducer = (state = initilState, action = {}) => {
    // if(action.type === '[TAREAS]agregar tarea'){
    //    return [...state, action.payload]
    // }

    switch (action.type) {
        case '[TAREAS]agregar tarea':
            return [...state, action.payload]

        case '[TAREAS]eliminar tarea':
            return state.filter(tarea => tarea.id !== action.payload.id)
        default:
            return state;
    }
}


// console.log(tareasReducer(initilState, addtarea));
// tareasReducer(initilState, addtarea)
export const ListTask = () => {


    const addtarea = (event) => {
        event.preventDefautl()
        // type: '[TAREAS]agregar tarea',
        //     payload: nuevatarea

    }

    const { tarea, formState, onInputChange } = useForm({ tarea: '' })
    return (
        <>
            <form onSubmit={addtarea}  >
                <div className="form-group">
                    <input
                        type="text"
                        name="tarea"
                        className="form-control"
                        placeholder="Ingresa tarea"
                        value={tarea}
                        onChange={onInputChange}
                    />
                </div>
                <button
                    type="subtmit"
                    className="btn btn-primary">
                    Submit
                </button>
            </form>
            <hr />
            <ul>
                {initilState.map(({id,tarea}) =><li key={id}>{tarea}{id}</li> )}

            </ul>
        </>
    )
}