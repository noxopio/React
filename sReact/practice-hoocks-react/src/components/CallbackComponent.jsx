import { useCallback, useState } from "react"
import { Incrementar } from "./Incrementar"

export const CallbackComponent = () => {


    const [contador, setContador] = useState(0)
    const incremetarPadre = useCallback((val)=>{
        setContador(cont => cont + val)
    },[])


    

    return (
        <>
            <h1>Contador: {contador}</h1>
            <Incrementar
                incrementar={incremetarPadre}
            />

        </>
    )
}
