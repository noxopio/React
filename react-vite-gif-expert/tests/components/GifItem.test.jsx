import { render ,screen} from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('GifItem', () => {

    const titulo = 'Alucard';
    const url = 'https://Alucard.com/alucar.jpg';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={titulo} url={url} />);
        expect(container).toMatchSnapshot();
    })

test('Debe tener un parrafo con el titulo', () => {
  
    const { getByText } = render(<GifItem title={titulo} url={url} />);
    expect(getByText(titulo)).toBeTruthy()







});











}); 