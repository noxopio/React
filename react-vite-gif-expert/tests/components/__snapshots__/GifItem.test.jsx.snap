// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Alucard"
      src="https://Alucard.com/alucar.jpg"
    />
    <p>
      Alucard
    </p>
  </div>
</div>
`;
